### PR TITLE
[FIX] Be robust to 4D T1w images

### LINKED ISSUE #792: t1_template_dimensions - TraitError
DS210 / sub-31

Seems like this T1w is 4D:

```Bash
$ nib-ls /work/03843/crn_plab/stampede2/fmriprep-phase2/data/openfmri/ds000210/sub-31/anat/sub-31_T1w.nii.gz
/work/03843/crn_plab/stampede2/fmriprep-phase2/data/openfmri/ds000210/sub-31/anat/sub-31_T1w.nii.gz float64 [256, 256, 176,   1] 1.00x1.00x1.00x0.00
```

```Python
Node: fmriprep_wf.single_subject_31_wf.anat_preproc_wf.t1_template_dimensions
Working directory: /scratch/03763/oesteban/fmriprep-phase2/work/ds000210/sub-31/fmriprep_wf/single_subject_31_wf/anat_preproc_wf/t1_template_dimensions

Node inputs:

ignore_exception = False
max_scale = 3.0
t1w_list = ['/work/03843/crn_plab/stampede2/fmriprep-phase2/data/openfmri/ds000210/sub-31/anat/sub-31_T1w.nii.gz']

Traceback (most recent call last):
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/plugins/multiproc.py", line 51, in run_node
    result['result'] = node.run(updatehash=updatehash)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 407, in run
    self._run_interface()
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 517, in _run_interface
    self._result = self._run_command(execute)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 650, in _run_command
    result = self._interface.run()
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base.py", line 1089, in run
    outputs = self.aggregate_outputs(runtime)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base.py", line 1176, in aggregate_outputs
    raise error
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base.py", line 1170, in aggregate_outputs
    setattr(outputs, key, val)
  File "/usr/local/miniconda/lib/python3.6/site-packages/traits/trait_handlers.py", line 172, in error
    value )
traits.trait_errors.TraitError: The 'target_shape' trait of a TemplateDimensionsOutputSpec instance must be a tuple of the form: (an integer (int or long), an integer (int or long), an integer (int or long)), 
but a value of (176, 256, 256, 1) <class 'tuple'> was specified.
```

### PR DESCRIPTION
**Pull request by oesteban**
Fixes #792

### CODE REVIEW

File: fmriprep/interfaces/images.py [hunk 1]
```diff
@@ -232,7 +232,7 @@ def _run_interface(self, runtime):
         target_span = target_shape * target_zooms
 
         zooms = np.array(reoriented.header.get_zooms()[:3])
-        shape = np.array(reoriented.shape)
+        shape = np.array(reoriented.shape[:3])
```
Comment (effigies) [inline fmriprep/interfaces/images.py]: If we don't need to resample 4D images, then the output of this node will still be a 4D image. Do the downstream nodes handle this okay?

Comment (oesteban) [reply fmriprep/interfaces/images.py]: This change is actually a downstream artifact of the change in line 166.

I'm testing this out right now, so far so good.

Comment (oesteban) [reply fmriprep/interfaces/images.py]: It went through the whole execution with ``--anat-only`` on the problematic dataset :+1:

Comment (effigies) [reply fmriprep/interfaces/images.py]: Sounds good.

**Pull request merged**